packages/manager/modules/pci/src/projects/project/ai-dashboard/components/ai-pipeline/ai-pipeline.controller.js

```diff
@@ -0,0 +1,78 @@
+import { TOOLS_STATUSES } from './ai-pipeline.constants';
+import exploreImage from './assets/explore.png';
+import serveImage from './assets/serve.png';
+import stockImage from './assets/stock.png';
+import trainImage from './assets/train.png';
+
+export default class AIPipelineCtrl {
+  /* @ngInject */
+  constructor($scope) {
+    this.$scope = $scope;
+  }
+
+  $onInit() {
+    // Compute statuses when items change
+    this.$scope.$watch(
+      '$ctrl.aiItems',
+      () => {
+        this.updateStatus();
+      },
+      true,
+    );
```

Comment:
**question:** Instead of adding a watch, cant we know when the `aiItems` is updated and then trigger a function to handle the changes?